pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -362,14 +376,33 @@ func (p *ConfigurationEditFlags) Apply(
 		}
 	}
 
-	if cmd.Flags().Changed("label") {
-		labelsMap, err := util.MapFromArrayAllowingSingles(p.Labels, "=")
+	if cmd.Flags().Changed("label") || cmd.Flags().Changed("label-service") || cmd.Flags().Changed("label-revision") {
+		serviceMap := make(util.StringMap)
+		revisionMap := make(util.StringMap)
+
+		labelsAllMap, err := util.MapFromArrayAllowingSingles(p.Labels, "=")
 		if err != nil {
 			return errors.Wrap(err, "Invalid --label")
 		}
+		serviceMap.Merge(labelsAllMap)
+		revisionMap.Merge(labelsAllMap)
+
+		labelServiceMap, err := util.MapFromArrayAllowingSingles(p.LabelService, "=")
```

Comment:
we could put the duplicate code into a single method called twice. This time we can _guarantee_ that the specific labels are treated the same.